### PR TITLE
adding IANA Root Zone Database collection

### PR DESCRIPTION
**Pull request by nyamsprod**
## Introduction

Adding IANA Root Zone Database

### Describe the new feature

Add the ability to resolve domain name against [IANA Root Zone Database](https://www.iana.org/domains/root/db)

```php
<?php

$manager = new Manager(new Cache(), new CurlHttpClient(), '1 DAY');
$tldList = $manager->getTLDs();
$tldList ->contains('brussels'); //return true;
$domain = $tldList ->resolve('ulb.ac.be');
$domain->getPublicSuffix(); //return 'be';
```

### Backward Incompatible Changes

None

### Targeted release version

next minor release

### PR Impact

adds methods to final classes

## Open issues

None

### CODE REVIEW

**Comment by jeremykendall** [comment]
@nyamsprod Checked this branch out so I could run the tests and I'm getting the following error when running `composer install`: 

```bash
> \Pdp\Installer::updateLocalCache
Updating your Public Suffix List local cache.
😓 😓 😓 Your local cache could not be updated. 😓 😓 😓
An error occurred during the update.
----- Error Message ----
SSL peer handshake failed, the server most likely requires a client certificate to connect
```

Any ideas?

**Comment by nyamsprod** [comment]
@jeremykendall this means that the default setup for the cURLHttpClient does not work for you. See
https://github.com/jeremykendall/php-domain-parser/blob/develop/src/CurlHttpClient.php#L30-L39

you can still install the package but you need to rewrite the update-psl script ... maybe this should be documented

**Comment by jeremykendall** [comment]
@nyamsprod Definitely want to document that. I just made a few changes to the adapter directly and got things updated.

File: tests/CacheTest.php [hunk 1]
```diff
@@ -58,30 +58,29 @@ public function tearDown()
     public function testConstructorOnEmptyCachePath()
     {
         $cache = new Cache('');
-        $this->assertNull($cache->get('invalid_key'));
+        self::assertNull($cache->get('invalid_key'));
```
[the remark this answers is not in the document]
Comment (nyamsprod) [reply tests/CacheTest.php]: I'll make a separate PR for that and revert to this for this PR

File: src/CurlHttpClient.php [hunk 2]
```diff
@@ -15,6 +15,20 @@
...
+use const CURLOPT_HTTPGET;
+use const CURLOPT_RETURNTRANSFER;
+use const CURLOPT_SSL_VERIFYHOST;
+use const CURLOPT_SSL_VERIFYPEER;
+use function curl_close;
+use function curl_errno;
+use function curl_error;
+use function curl_exec;
+use function curl_init;
+use function curl_setopt_array;
```
[the remark this answers is not in the document]
Comment (nyamsprod) [reply src/CurlHttpClient.php]: Sorry that my IDE settings ... I'll revert. It's also a micro optimisation for php which bypass looking into the current namespace for global functions

**Comment by nyamsprod** [comment]
Classes with code handling both PSL and RZD should ideally be refactored so that responsibility for those different concerns are handled in smaller classes. Thinking specifically of Manager. 
Once I remove the extra code you'll see that the Manager responsibilities are well defined ... I thought having 2 manager classes would have been overkill since in both cases the logic is the same the difference is on the conversion algorithms. And each algorithm has its own class... The Converter ones.

**Comment by nyamsprod** [comment]
@jeremykendall I've reverted all the "cosmetic changes" now the PR should be cleaner.

**Pull request merged**